designsafe/static/scripts/rapid/controllers/rapid-main-ctrl.js

```diff
@@ -47,51 +60,193 @@ export default class RapidMainCtrl {
 
-    select_event (ev) {
+    select_event(ev) {
         this.show_sidebar = true;
+        let lat = ev.location ? ev.location.lat : ev.geometry.coordinates[1];
+        let lon = ev.location ? ev.location.lon : ev.geometry.coordinates[0];
         if (ev === this.active_rapid_event) {
             this.reset();
         } else {
-            this.map.setView([ev.location.lat, ev.location.lon],8, { animate: true });
+            this.map.setView([lat, lon], 8, { animate: true });
             this.active_rapid_event = ev;
-            this.$location.search('event', ev.title);
+            this.$location.search('event', ev.title || ev.properties && ev.properties.name);
+
+            if (ev.properties && ev.properties.host === 'OpenTopo') {
+                this.RapidDataService.get_opentopo_coordinates(ev.properties.doiUrl).then((data) => {
+                    if (this.active_polygon) {
+                        this.map.removeLayer(this.active_polygon);
+                    }
+                    this.active_polygon = this.addPolygonToMap(data.geometry, ev.properties);
+                }).catch((err) => {
+                    console.error('Error loading OpenTopography coordinates:', err);
+                });
+            }
+        }
+    }
+
+    addPolygonToMap(geometry, properties) {
+        let coordinates = geometry.coordinates;
+        let polygonLayer;
+
+        let popupContent = `<b>${properties.name}</b><br>
+                            <b>Dataset Name:</b> ${properties.name}<br>
+                            <b>Data Source:</b> Opentopography <br>
+                            <b>Products available:</b> ${properties.productAvailable}<br>
+                            <b>Survey date:</b> ${properties.temporalCoverage}<br>
+                            <b>DOI:</b> <a href="${properties.doiUrl}" target="_blank">${properties.doiUrl}</a>`;
+
+        if (geometry.type === 'MultiPolygon') {
+            polygonLayer = L.layerGroup();
+            coordinates.forEach(singlePolygon => {
+                let latlngs = singlePolygon[0].map(coord => [coord[1], coord[0]]);
+                let polygon = L.polygon(latlngs, { color: '#a8bb2f' });
+                polygon.bindPopup(popupContent);
+                polygon.addTo(polygonLayer);
+            });
+            polygonLayer.addTo(this.map);
+        } else if (geometry.type === 'Polygon') {
+            let latlngs = coordinates[0].map(coord => [coord[1], coord[0]]);
+            polygonLayer = L.polygon(latlngs, { color: '#a8bb2f' });
+            polygonLayer.bindPopup(popupContent);
+            polygonLayer.addTo(this.map);
         }
+
+        return polygonLayer;
     }
 
-    search () {
-        this.filtered_events = this.RapidDataService.search(this.events, this.filter_options);
+    search() {
+        if (!this.events || !this.openTopoData) return;
+        this.filtered_events = this.RapidDataService.search(this.events.concat(this.openTopoData), this.filter_options);
```

Comment:
this is failing here as we are searching on `title`

Reply:
Removed the url search param for opentopo events. Now it will not add any url params for opentopo markers on click- as it is creating complications related to opentopo data not being shown default and as well - if no cache found it would break this functionality. 
Also made a fix to existing events data- clearing url search params- which was present before.

Reply:
Thanks Nathan, for the detail.
Fixed it, removed the search capabilities and listing of opentopo data from the listing.
Filtering is being addressed in the task- https://tacc-main.atlassian.net/browse/DES-3006

Reply:
If the opentopo data is completely removed from the listing, then filtering will not be possible and will break the code, because events and opentopoData are 2 different listings.

Reply:
I think it should be fixed by adding the below line in the filters branch to populate the opentopo list data.
`<event-listing ng-repeat="event in vm.openTopoData" event=event ng-click="vm.select_event(event)"></event-listing>`